core/domain/wipeout_service.py

```diff
@@ -0,0 +1,119 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Services for user data."""
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from constants import constants
+from core.domain import collection_services
+from core.domain import exp_fetchers
+from core.domain import exp_services
+from core.domain import skill_services
+from core.domain import topic_services
+from core.domain import user_services
+from core.platform import models
+
+current_user_services = models.Registry.import_current_user_services()
+(user_models,) = models.Registry.import_models([models.NAMES.user])
+transaction_services = models.Registry.import_transaction_services()
+
+
+def _activity_should_be_deleted(activity_summary, user_id):
+    """Return if activity should be deleted, thus if it is private and the user
+    is the sole owner of the activity.
+
+    Args:
+        activity_summary: ExplorationSummary|CollectionSummary. The activity
+            summary.
+        user_id: str. The id of the user to be deleted.
+
+    Returns:
+        bool. Whether the activity should be deleted.
+    """
+    return (
+        activity_summary.status == constants.ACTIVITY_STATUS_PRIVATE and
+        user_id in activity_summary.owner_ids and
+        len(activity_summary.owner_ids) == 1
+    )
+
+
+def _topic_should_be_deleted(topic_rights, user_id):
+    """Return if topic should be deleted, thus if it is not published and
+    the user is the sole owner of the topic.
+
+    Args:
+        topic_rights: TopicRights. The topic rights.
+        user_id: str. The id of the user to be deleted.
+
+    Returns:
+        bool. Whether the topic should be deleted.
+    """
+    return (
+        not topic_rights.topic_is_published and
+        topic_rights.is_manager(user_id) and
+        len(topic_rights.manager_ids) == 1
+    )
+
+
+def pre_delete_user(user_id):
+    """Prepare user for the full deletion
```

Comment:
End sentence with a period.

Please also give more detail here about what this preparation entails.